CMakeLists.txt

```diff
@@ -13,7 +13,9 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-cmake_minimum_required (VERSION 3.9)
+cmake_minimum_required (VERSION 3.12)
+
+enable_language(CUDA)
```

Comment:
I don't see anything in this PR that makes CUDA optional as we discussed.

Reply:
Yeah. I haven't accomplished this function yet. This version is just hard coded the bool value in the header file as GPU = 1. Will change it soon!